File: src/main/java/quizapi/repository/GameSessionRepository.java
package quizapi.repository;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import quizapi.modal.Category;
import quizapi.modal.GameSession;
import quizapi.modal.User;

import java.util.Optional;

@Repository
public interface GameSessionRepository extends CrudRepository<GameSession, Long> {
    Optional<GameSession> findByUser(User user);
}

File: src/main/kotlin/com/shkila/quizapi/repository/GameSessionRepository.kt
package com.shkila.quizapi.repository

import com.shkila.quizapi.model.GameSession
import com.shkila.quizapi.model.User
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface GameSessionRepository : CrudRepository<GameSession, Long> {
    fun findByUser(user: User): Optional<GameSession>
}
